azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VirtualMachineApi.java

```diff
@@ -0,0 +1,140 @@
+import org.jclouds.rest.annotations.QueryParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.net.URI;
+import java.util.List;
+
+/**
+ * The Virtual Machine API includes operations for managing the virtual machines in your subscription.
+ *
+ * @see <a href="https://msdn.microsoft.com/en-us/library/azure/mt163630.aspx">docs</a>
+ */
+@Path("/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/virtualMachines")
+@RequestFilters(OAuthFilter.class)
+@QueryParams(keys = "api-version", values = "2015-06-15")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface VirtualMachineApi {
+
+   /**
+    * The Get Virtual Machine details
+    */
+   @Named("GetVirtualMachine")
+   @GET
+   @Path("/{name}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   VirtualMachine get(@PathParam("name") String name);
+
+   /**
+    * The Get Virtual Machine details
+    */
+   @Named("GetVirtualMachineInstance")
+   @GET
+   @Path("/{name}/instanceView")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   VirtualMachineInstance getInstanceDetails(@PathParam("name") String name);
+
+   /**
+    * The Create Virtual Machine
+    */
+   @Named("CreateVirtualMachine")
+   @PUT
+   @Payload("%7B\"id\":\"{id}\",\"name\":\"{name}\",\"type\":\"Microsoft.Compute/virtualMachines\"," +
+           "\"location\":\"{location}\",\"tags\":%7B%7D,\"properties\":{properties}%7D")
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/{vmname}")
+   @QueryParams(keys = "validating", values = "false")
+   VirtualMachine create(@PathParam("vmname") String vmname, @PayloadParam("id") String id,
+                         @PayloadParam("name") String name,
+                         @PayloadParam("location") String location,
+                         @PayloadParam("properties") VirtualMachineProperties properties);
+
+   /**
+    * The List Virtual Machines operation
+    */
+   @Named("ListVirtualMachines")
+   @GET
+   @SelectJson("value")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<VirtualMachine> list();
+
+   /**
+    * The Delete Virtual Machine operation
+    */
+   @Named("DeleteVirtualMachine")
+   @DELETE
+   @Path("/{name}")
+   @ResponseParser(URIParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   URI delete(@PathParam("name") String name);
+
+   /**
+    * The Restart Virtual Machine operation
+    */
+   @Named("RestartVirtualMachine")
+   @POST
+   @Path("/{name}/restart")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
```

Comment:
We don't want to silently succeed if the VM does not exist. Remove the 404 fallbacks in all state change operations.